pandas/core/groupby/groupby.py

```diff
@@ -624,7 +624,10 @@ def _get_index(self, name):
         """
         Safe get index, translate keys for datelike to underlying repr.
         """
-        return self._get_indices([name])[0]
+        if isna(name):
+            return [i for i, v in enumerate(self.indices) if isna(v)]
+        else:
+            return self._get_indices([name])[0]
```

Comment:
IIUC shouldn't we just fix get_indices to properly handle NA?

Reply:
I'm down to do that - will have a go!